pkg/util/ip_reservation_test.go

```diff
@@ -0,0 +1,211 @@
+		if err != nil && !test.errorExpected {
+			t.Errorf("test %q failed: got error %s, expected %s", test.name, err.Error(), test.expected)
+		} else if err == nil && test.errorExpected {
+			t.Errorf("test %q failed: got reserved IP range %s, expected error", test.name, ipBlock)
+		} else if ipBlock != test.expected {
+			t.Errorf("test %q failed: got reserved IP range %s, expected %s", test.name, ipBlock, test.expected)
+		}
+	}
+}
+func TestValidateCIDROverlap(t *testing.T) {
+	cases := []struct {
+		name          string
+		cidr1         string
+		cidr2         string
+		expected      bool
+		errorExpected bool
+	}{
+		{
+			name:          "Overlapping CIDRs",
+			cidr1:         "192.168.92.0/29",
+			cidr2:         "192.168.92.48/26",
+			expected:      true,
+			errorExpected: false,
+		},
+		{
+			name:          "Non overlapping CIDRs",
+			cidr1:         "192.168.92.0/29",
+			cidr2:         "192.168.22.67/26",
+			expected:      false,
+			errorExpected: false,
+		},
+		{
+			name:          "Invalid CIDR provided",
+			cidr1:         "192.168.92.0",
+			cidr2:         "192.168.22.67/26",
+			errorExpected: true,
+		},
+	}
+
+	for _, test := range cases {
+		overlap, err := isOverlap(test.cidr1, test.cidr2)
+		if err != nil && !test.errorExpected {
+			t.Errorf("test %q failed: got error %s, expected cidr overlap between %s and %s to be %t", test.name, err.Error(), test.cidr1, test.cidr2, test.expected)
+		} else if err == nil && test.errorExpected {
+			t.Errorf("test %q failed: got cidr overlap value %t, expected error", test.name, overlap)
+		} else if !test.errorExpected && overlap != test.expected {
+			t.Errorf("test %q failed: got overlap for cidr %s and %s as %t, expected %t", test.name, test.cidr1, test.cidr2, test.expected, test.expected)
+		}
+	}
+}
+
+func TestIncrementIP(t *testing.T) {
+
+	cases := []struct {
+		name          string
+		currentIP     string
+		step          byte
+		expected      string
+		errorExpected bool
+	}{
+		{
+			name:          "Valid IP increment without carry forward to significant bytes",
+			currentIP:     "192.168.92.32",
+			step:          7,
+			expected:      "192.168.92.39",
+			errorExpected: false,
+		},
+		{
+			name:          "Valid IP increment without carry forward to significant bytes",
+			currentIP:     "192.168.92.32",
+			step:          143,
+			expected:      "192.168.92.175",
+			errorExpected: false,
+		}, {
+			name:          "Valid increment with carry forward to significant bytes",
+			currentIP:     "192.168.255.253",
+			step:          7,
+			expected:      "192.169.0.4",
+			errorExpected: false,
+		}, {
+			name:          "Valid increment with carry forward to significant bytes",
+			currentIP:     "192.255.255.253",
+			step:          255,
+			expected:      "193.0.0.252",
+			errorExpected: false,
+		},
+		{
+			name:          "Invalid increment",
+			currentIP:     "255.255.255.253",
+			step:          167,
+			errorExpected: true,
+		},
```

Comment:
add test

```
currentIP:     "0.255.255.255",
step:          3,
expected: "1.0.0.2",
```

Reply:
I think `currentIP:     "192.255.255.253",
			step:          255,
			expected:      "193.0.0.252",` is a similar case. But can add this if you want